src/app/login/login.service.ts

```diff
@@ -32,8 +32,12 @@ export class LoginService {
           observable.next(user);
           observable.complete();
         },
-        (error: HttpErrorResponse) => {
-          this.alertService.detailedError(error);
+        (error) => {
+          if (error.status !== undefined) {
```

Comment:
Why not just `if (error.status)`? You're confident it will never be null?

Reply:
We have code that checks if `error.status` is 0, `if (error.status)` would not trigger on it.
https://github.com/dockstore/dockstore-ui2/blob/9322ad39f1c37046b019a40c9933618b63b6e9ff/src/app/shared/alert/state/alert.service.ts#L91
The goal of the code is to figure out if the error object is an `HttpErrorResponse`, in which case `status` will very likely not be `null`.

Reply:
Postscript: what this code is trying to do is take one action if `error` is an `HttpErrorResponse`, and a different action otherwise.  Right now, it uses the presence of `status` as a proxy for `HttpErrorResponse`-ness, but maybe there's a better way to do it in Typescript?  Here's a stackoverflow thread on how to determine if an object has a certain property, the `in` solution of the first answer might be better... https://stackoverflow.com/questions/1894792/how-to-determine-whether-an-object-has-a-given-property-in-javascript